Mathlib/RingTheory/LocalProperties/Reduced.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2021 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.RingTheory.LocalProperties.Basic
+
+/-!
+# `IsReduced` is a local property
+
+In this file, we prove that `IsReduced` is a local property.
+
+
+## Main results
+
+Let `R` be a commutative ring, `M` be a submonoid of `R`.
+
+* `localization_isReduced` :  `M⁻¹R` is reduced if `R` is reduced.
+* `isReduced_ofLocalizationMaximal` : `R` is reduced if `Rₘ` is reduced for all maximal ideal `m`.
+
+-/
+
+/-- `M⁻¹R` is reduced if `R` is reduced. -/
+theorem localization_isReduced : LocalizationPreserves fun R hR => IsReduced R := by
+  introv R _ _
+  constructor
+  rintro x ⟨_ | n, e⟩
```

Comment:
We should fix the naming of these at some point. It should be called `isReduced_localizationPreserves`, but this can be in a later PR.